proposals/NNNN-Add-screen-stack-management-module.md

```diff
@@ -0,0 +1,96 @@
+# Add screen stack management module
+
+* Proposal: [SDL-NNNN](NNNN-Add-screen-stack-management-module.md)
+* Author: [Akihiro Miyazaki (Nexty)](https://github.com/Akihiro-Miyazaki)
+* Status: **Awaiting review**
+* Impacted Platforms: [ Core / iOS / Java Suite / JavaScript Suite / RPC ]
+
+## Introduction
+The purpose of this proposal is to reduce the burden on screen development for app developers by adding a screen management module.
+
+
+## Motivation
+Currently, the screens displayed by `Show` and `PerformInteraction` are independent, and there is no relation between each screen. However, in many cases of the App development process, each screen has a relationship with each other. Therefore, the screen transition specifications expected by the app developers may not be realized. By adding a screen management module and managing transitions, this problem can be solved and reduce the burden to screen development. For example, pressing the back button to enable screen transitions such as return to the previous screen. As a result, it is possible to build a better UX by reducing the inconvenience of use as much as possible for the users.
+
+<b>Current behavior</b>
+ - `PerformInteraction` and `Show (template)` are independent screens.
+ - HMI controls menu/submenu displays and screen transitions.
+ - Contents displayed in menu/submenu are provided by `AddCommand`, `DeleteCommand`, `AddSubMenu`, `DeleteSubMenu` of the app.
+ - When pushing the menu item, the HMI sends an `onCommand` notification to the mobile, and then the mobile determines the operation.
+   After that, you cannot return to the menu screen, but you will return to the first template screen.
+ - Even if the `DeleteSubMenu` is notified from mobile while the submenu is displayed, it is rejected.
+
+<b>Current Screen transition process</b>
+
+- RPC related to menu/submenu are sent after the successful registration of app.
+```
+AddCommand(commandID = 1，"command1")
+AddSubMenu(menuID = 100, "submenu1")
+AddCommand(commandID = 2，"command2")
+AddCommand(commandID = 4，"command4", parentID = 100)
+AddCommand(commandID = 5，"command5", parentID = 100)
+AddCommand(commandID = 6，"command6", parentID = 100)
+```
+
+- RPC for `PerformInteraction`
+```
+CreateInteractionChoiceSet[
+	 interactionChoiceSetID = 1000,
+	(choiceID = 1,  menuName = "choice1"),
+	(choiceID = 2,  menuName = "choice2"),
+	(choiceID = 3,  menuName = "choice3") ]
+```
+
+![Screen_transition_process.png](../assets/proposals/NNNN-Add-screen-stack-management-module/Screen_transition_process.png)
```

Comment:
There's a typo in this image that needs to be fixed: "mene" should be "menu".  Additionally, "proxy" should be changed to "SDL App Library".

Reply:
Hi, @theresalech -san.
Thank you in advance for your prompt reply.
I will modify the image file and make a commit it.
Could you please confirm it again.
Thank you.